src/main/java/Naega.java

```diff
@@ -0,0 +1,150 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Naega {
+    public static void main(String[] args) {
+
+        // Greeting the user
+        System.out.println("____________________________________________________________");
+        System.out.println(" Hello! I'm Naega");
+        System.out.println(" What can I do for you?");
+        System.out.println("____________________________________________________________");
+
+        // Initialize a list to store user tasks
+        ArrayList<Task> tasks = new ArrayList<>();
+        Scanner scanner = new Scanner(System.in);
+        String userInput;
+
+        // Main loop
+        while (true) {
+            userInput = scanner.nextLine();
+
+            try {
+                if (userInput.equals("bye")) {
+                    System.out.println("____________________________________________________________");
+                    System.out.println(" Bye. Hope to see you again soon!");
+                    System.out.println("____________________________________________________________");
+                    break;
+                } else if (userInput.equals("list")) {
+                    handleList(tasks);
+                } else if (userInput.startsWith("mark ")) {
+                    handleMark(tasks, userInput);
+                } else if (userInput.startsWith("unmark ")) {
+                    handleUnmark(tasks, userInput);
+                } else if (userInput.startsWith("todo ")) {
+                    handleTodo(tasks, userInput);
+                } else if (userInput.startsWith("deadline ")) {
+                    handleDeadline(tasks, userInput);
+                } else if (userInput.startsWith("event ")) {
+                    handleEvent(tasks, userInput);
+                } else if (userInput.startsWith("delete ")) {
+                    handleDelete(tasks, userInput);
+                } else {
+                    throw new NaegaException("I don't know how to handle that command.");
+                }
+            } catch (NaegaException e) {
+                System.out.println("____________________________________________________________");
+                System.out.println(" Oops! " + e.getMessage());
+                System.out.println("____________________________________________________________");
+            } catch (NumberFormatException | IndexOutOfBoundsException e) {
+                System.out.println("____________________________________________________________");
+                System.out.println(" Oops! Invalid task number.");
+                System.out.println("____________________________________________________________");
+            }
```

Comment:
I like how mostly clean this block of code is! Any reason for using if-else instead of switch-case here though?